src/lib.rs

```diff
@@ -195,45 +195,51 @@ fn get_num_physical_cpus_windows() -> Option<usize> {
 
 #[cfg(target_os = "linux")]
 fn get_num_physical_cpus() -> usize {
-    use std::io::BufReader;
-    use std::io::BufRead;
+    use std::collections::HashMap;
     use std::fs::File;
-    use std::collections::HashSet;
+    use std::io::BufRead;
+    use std::io::BufReader;
 
     let file = match File::open("/proc/cpuinfo") {
         Ok(val) => val,
-        Err(_) => {return get_num_cpus()},
+        Err(_) => return get_num_cpus(),
     };
     let reader = BufReader::new(file);
-    let mut set = HashSet::new();
-    let mut coreid: u32 = 0;
+    let mut map = HashMap::new();
     let mut physid: u32 = 0;
+    let mut cores: usize = 0;
     let mut chgcount = 0;
     for line in reader.lines().filter_map(|result| result.ok()) {
-        let mut it = line.split(':');
-        let (key, value) = match (it.next(), it.next()) {
-            (Some(key), Some(value)) => (key.trim(), value.trim()),
-            _ => continue,
-        };
-        if key == "core id" || key == "physical id" {
-            let value = match value.parse() {
-              Ok(val) => val,
-              Err(_) => break,
+        let parts: Vec<&str> = line.split(':').map(|s| s.trim()).collect();
```

Comment:
It looks like this is reverting a recent change to just use the iterator instead of allocating a new Vec.

Reply:
Damn. That's what you get by doing the fix with the OpenBSD travel laptop in the middle of the company offsite. Fixed.